src/mongo_client.ts

```diff
@@ -433,6 +434,18 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> {
 
     const options = this[kOptions];
 
+    if (options.tls) {
+      if (!options.ca && typeof options.tlsCAFile === 'string' && options.tlsCAFile.length > 0) {
+        options.ca = await fs.readFile(options.tlsCAFile, { encoding: 'utf8' });
+      }
```

Comment:
I have two related cleanups here (these apply to `tlsCertificateKeyFile` too)

first, we can use the nullish coelescing assignment operator here to avoid checking `!options.ca`:
```typescript
      if (typeof options.tlsCAFile === 'string' && options.tlsCAFile.length > 0) {
        options.ca ??= await fs.readFile(options.tlsCAFile, { encoding: 'utf8' });
      }
```

---

second, I'm not sure we should be checking `options.tlsCAFile.length` (even though the AC says to).  the user must manually provide the filepath, and if we receive an empty string, the user's configuration is likely messed up.  We should probably alert the user that we received an empty file path instead of silently ignoring it.  Attempting to read from a file named "" will throw an error, so that was the old behavior.  

If we do decide to ignore empty strings, I suggest we move the logic into the options parser.  The requirement for these options would actually be "a non-empty string", instead of just "a string", so that should be codified in the options parsing and tested there.

Reply:
Your suggestion to handle empty file names in option parsing is more user friendly as far as catching bad config is concerned, so that's the solution I'd prefer as well.

Reply:
Daria wrote the ticket and included "ignore empty filenames" as an AC.  Can you ask her why that was included?

Reply:
IIRC we wanted to have a nicer experience for environment variables being the source of the file name which can be set to an empty string as a way of clearing the value. I do not feel strongly, if we do ignore the empty string wouldn't TLS be misconfigured and fail? Is there a scenario where a server with TLS configured allows a client without TLS to connect? Also happy with keeping the behavior as it was (validate string and not length)

Reply:
Well, in both cases it fails to connect, but if we throw ourselves it's a much more actionable error than a generic connection error. 

Did we have a specific environment / usecase / testing variant we wanted this for?

Reply:
> Is there a scenario where a server with TLS configured allows a client without TLS to connect?

Not sure what impact that has here, but sure, the server’s TLS mode can be `disabled`, `allowTLS`, `preferTLS`, `requireTLS`, and two of these are modes where the server accepts connections with or without TLS.